app/components/primer/alpha/action_list_item.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    class ActionListItem < Primer::Component
+      status :alpha
+
+      DEFAULT_SIZE = :small
+      SIZE_MAPPINGS = {
+        DEFAULT_SIZE => nil,
+        :medium => "ActionList-content--sizeMedium",
+        :large => "ActionList-content--sizeLarge"
+      }.freeze
+      SIZE_OPTIONS = SIZE_MAPPINGS.keys.freeze
+
+      DEFAULT_DESCRIPTION_VARIANT = :block
+      DESCRIPTION_VARIANT_MAPPINGS = {
+        :inline => "ActionList-item-descriptionWrap--inline",
+        DEFAULT_DESCRIPTION_VARIANT => "ActionList-item-descriptionWrap--block"
+      }.freeze
+      DESCRIPTION_VARIANT_OPTIONS = DESCRIPTION_VARIANT_MAPPINGS.keys.freeze
+
+      DEFAULT_VARIANT = :default
+      VARIANT_MAPPINGS = {
+        DEFAULT_VARIANT => nil,
+        :danger => "ActionList-item--danger"
+      }.freeze
+      VARIANT_OPTIONS = VARIANT_MAPPINGS.keys.freeze
+
+      DEFAULT_SELECT_MODE = :none
```

Comment:
I'm wondering if we might be able to remove this select mode check. For single select, we'll want `aria-selected="true"` on the `li`, and for multi `aria-checked="true"`. So maybe it can be either `checked` or `selected`?

Additionally, we should make sure `aria-etc` is only set when these are true and removed from the dom when `false`. Its just extra noise we don't need.

Reply:
Hmm ok I see what you mean. Is the idea then that `checked` would imply a checkbox and `selected` a check mark?

Reply:
It now occurs to me you were actually saying we might not need the `:none` case. I agree and will remove it.

Reply:
Okay, I think having the select mode option is correct. We do need a `:none` case, but that should be default. So you should be able to opt into single or multi select. I think the none state isn't working right now.